### PR TITLE
Refactor: plugin jenkins-pipeline Config Options Refactoring

### LINKED ISSUE #1055: :four_leaf_clover: `Proposal`: `jenkins-pipeline` Options Redesign
## What Would You Like to Add? Why Is This Needed?

`jenkins-pipeline` options redesign

## Design

- Old

```yaml
tools:
- name: jenkins-pipeline
  instanceID: default
  dependsOn: []
  options:
    jenkinsURL: http://54.71.232.26:32000
    jenkinsUser: admin
    jenkinsfilePath: https://xxx/Jenkinsfile
    jenkinsEnableRestart: true
    projectURL: http://54.71.232.26:30080/root/spring-demo
    jenkinsNamespace: jenkins
    harborURL: http://harbor.example.com:80
    harborUser: admin
```

- New

```yaml
tools:
- name: jenkins-pipeline
  instanceID: default
  dependsOn: []
  options:
    jenkins:
      url: http://54.71.232.26:32000/
      user: admin
      namespace: jenkins
      enableRestart: true
    scm:
      projectURL: http://54.71.232.26:30080/root/spring-demo
      projectBranch: master
    pipeline:
      jobName: jobs/test-job
      jenkinsfilePath: https://xxx/Jenkinsfile
      imageRepo:
        url: http://harbor.example.com:80
        user: admin
```

### PR DESCRIPTION
**Pull request by daniel-hutao**
## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [ ] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [ ] My code has the necessary comments and documentation (if needed).
- [ ] I have added relevant tests

## Description

plugin jenkins-pipeline config options refactoring. see #1055 for more info

## Related Issues

close #1055

## New Behavior (screenshots if needed)

test:

- config

```yaml
---
# core config
varFile: "" # If not empty, use the specified external variables config file
toolFile: "" # If not empty, use the specified external tools config file
state: # state config, backend can be local or s3
  backend: local
  options:
    stateFile: devstream.state

---
tools:
- name: jenkins-pipeline
  instanceID: default
  dependsOn: []
  options:
    jenkins:
      # url is used to config jenkins url
      url: http://54.71.232.26:32000
      # jenkins' user name
      user: admin
      # jenkins namespace in k8s cluster
      namespace: jenkins
      # restart jenkins if true for plugin install
      enableRestart: true
    scm:
      # projectURL is the project repo location
      projectURL: http://54.71.232.26:30080/root/spring-demo
      # project branch, master as default
      projectBranch: master
    pipeline:
      # jobName is jenkins's job name; <jobFolder/jobName> or <jobName>; e.g. jobs/test-job, test-job, jobs2/test-job
      jobName: test-job
      # jenkinsfilePath is the location describe how to get Jenkinsfile, it can be remote or local
      jenkinsfilePath: https://raw.githubusercontent.com/devstream-io/dtm-jenkins-pipeline-example/main/springboot/Jenkinsfile
      imageRepo:
        # image repo URL for pulling/pushing
        url: http://harbor.example.com:80
        # image repo user name
        user: admin
```

- apply (`dtm apply -f config-jenkins-pipeline.yaml -y`)

```sh
2022-09-04 11:53:51 ℹ [INFO]  Apply started.
2022-09-04 11:53:51 ℹ [INFO]  Using dir </root/.devstream/plugins> to store plugins.
2022-09-04 11:53:51 ℹ [INFO]  Using local backend. State file: devstream.state.
2022-09-04 11:53:51 ℹ [INFO]  Tool (jenkins-pipeline/default) found in config but doesn't exist in the state, will be created.
2022-09-04 11:53:51 ℹ [INFO]  Start executing the plan.
2022-09-04 11:53:51 ℹ [INFO]  Changes count: 1.
2022-09-04 11:53:51 ℹ [INFO]  -------------------- [  Processing progress: 1/1.  ] --------------------
2022-09-04 11:53:51 ℹ [INFO]  Processing: (jenkins-pipeline/default) -> Create ...
2022-09-04 11:53:51 ℹ [INFO]  Secret jenkins/docker-config has been created.
2022-09-04 11:53:54 ✔ [SUCCESS]  Tool (jenkins-pipeline/default) Create done.
2022-09-04 11:53:54 ℹ [INFO]  -------------------- [  Processing done.  ] --------------------
2022-09-04 11:53:54 ✔ [SUCCESS]  All plugins applied successfully.
2022-09-04 11:53:54 ✔ [SUCCESS]  Apply finished.
```

### CODE REVIEW

File: internal/pkg/plugininstaller/jenkins/option.go [hunk 1]
```diff
@@ -44,6 +34,29 @@ type JobOptions struct {
...
+
+type Pipeline struct {
+	JobName         string    `mapstructure:"jobName" validate:"required"`
+	JenkinsfilePath string    `mapstructure:"jenkinsfilePath" validate:"required"`
+	ImageRepo       ImageRepo `mapstructure:"imageRepo"`
+}
+
+type ImageRepo struct {
+	URL  string `mapstructure:"url" validate:"url"`
+	User string `mapstructure:"User"`
```
Comment (iyear) [inline internal/pkg/plugininstaller/jenkins/option.go:57]: typo: `User` -> `user`

File: internal/pkg/plugininstaller/jenkins/validate.go [hunk 2]
```diff
@@ -73,7 +96,7 @@ func buildAdminToken(userName string) (*jenkins.BasicAuth, error) {
 			Password: jenkinsPassword,
 		}, nil
 	}
-	// 2. if not set, get user and password from secret
+	// 2. if not set, get User and password from secret
```
Comment (iyear) [inline internal/pkg/plugininstaller/jenkins/validate.go:99]: why is `User` and not `user`

**Review by iyear — APPROVED** [review]
LGTM 👏

File: internal/pkg/plugininstaller/jenkins/validate.go (hunk 2 above)
Comment (daniel-hutao) [reply internal/pkg/plugininstaller/jenkins/validate.go:99]: It's "user" here, changed by ide refactor

**Pull request merged**